### PR TITLE
Stack network stage 2

### PR DESCRIPTION
**Pull request by freignat91**
#280

Be able to create custom networks and associate stack services to them in order to establish private communication between services from separate stacks.

test:
- stop amp:  sudo ./swarm stop
- pull images: ./swarm pull
- build amp: glide cc && rm -rf vendor && glide install && make install
- start amp: sudo ./swarm start
- create stack test1: amp stack up test1 -f api/rpc/stack/test_samples/sample-10-1.yml
- create stack test2: amp stack up test2 -f api/rpc/stack/test_samples/sample-10-2.yml
- verify:
  - stack test1 and test2 are up and running using: ./swarm monitor
  - custom network "amp_net" is created, using: docker network ls
  - container test1.pinger is able to ping test2.pinger
    - docker ps | grep test1.pinger  -> get the container id [containerId1]
    - docker exec -it [containerId1] sh
    - inside the container:  ping test2-pinger  -> should return a valid address
  - container test2.pinger is able to ping test1.pinger
    - docker ps | grep test2.pinger  -> get the container id [containerId2]
    - docker exec -it [containerId2] sh
    - inside the container:  ping test1-pinger  -> should return a valid address

Regression tests:
- make test

minor:
- add volumes in service definition and in stack file (tests added in `make test`) #299
- add args in services definition and in stack file (tests added in `make test`)

### CODE REVIEW

File: api/rpc/service/service.go [hunk 1]
```diff
@@ -132,10 +138,51 @@ func (s *Service) Create(ctx context.Context, req *ServiceCreateRequest) (*Servi
...
+		if len(list) == 1 {
+			//create a anonymous  volume
+			fmt.Println("create anonymous volume")
+			service.TaskTemplate.ContainerSpec.Mounts = append(service.TaskTemplate.ContainerSpec.Mounts, mount.Mount{
+				Type: mount.TypeVolume,
+				Target: vol,
+			})
+		} else if len(list) == 2 {
+			if strings.Index(list[0],"/") < 0 {
+				//create names volume
```
Comment (ndegory) [inline api/rpc/service/service.go]: s/names/named

File: api/rpc/service/service.go [hunk 2]
```diff
@@ -132,10 +138,51 @@ func (s *Service) Create(ctx context.Context, req *ServiceCreateRequest) (*Servi
...
+				fmt.Println("create named volume"+list[0])
```
Comment (ndegory) [inline api/rpc/service/service.go]: s/volume"/volume "/

File: api/rpc/service/service.go [hunk 3]
```diff
@@ -132,10 +138,51 @@ func (s *Service) Create(ctx context.Context, req *ServiceCreateRequest) (*Servi
...
+	service.TaskTemplate.ContainerSpec.Mounts = []mount.Mount{}
+	for _, vol := range mounts {
+		list := strings.Split(vol, ":")
+		ro := false
+		if len(list) == 3 && list[2] == "ro" {
+			ro = true
+		}
+		if len(list) == 1 {
+			//create a anonymous  volume
+			fmt.Println("create anonymous volume")
```
Comment (ndegory) [inline api/rpc/service/service.go:156]: you mean "implicit bind type volume?"

**Review by ndegory — CHANGES_REQUESTED** [review]
small fixes, mostly in comments

File: api/rpc/service/service.go (hunk 3 above)
Comment (freignat91) [reply api/rpc/service/service.go:156]: A volume create by the container with no name and not bind to a host directory

File: api/rpc/service/service.go (hunk 2 above)
Comment (freignat91) [reply api/rpc/service/service.go]: done

File: api/rpc/service/service.go (hunk 1 above)
Comment (freignat91) [reply api/rpc/service/service.go]: done

**Review by generalhenry — APPROVED** [review]
Tested locally

File: api/rpc/service/service.go (hunk 3 above)
Comment (ndegory) [reply api/rpc/service/service.go:156]: from what I see it's still bind to a host directory, it's just implicit (source is same as the target)

**Review by ndegory — APPROVED** [review]

File: api/rpc/service/service.go (hunk 3 above)
Comment (freignat91) [reply api/rpc/service/service.go:156]: yes, no explicite bind.

**Comment by subfuzion** [comment]
Works great! LGTM 👍

**Pull request merged**